c7n/resources/appelb.py

```diff
@@ -266,21 +269,35 @@ class WafV2Enabled(Filter):
                   - type: wafv2-enabled
                     state: false
                     web-acl: testv2
+
+              - name: filter-wafv2-elb-regex
+                resource: app-elb
+                filters:
+                  - type: wafv2-enabled
+                    state: false
+                    web-acl: .*FMManagedWebACLV2-?FMS-.*
```

Comment:
the use case seems like a better fit for the fnmatch module versus using a full regex, which would allow for a simpler glob style expression to match on acl, wdyt?

Reply:
Hi @kapilt, I think using regex is more flexible than glob style matching. For example,  .*FMManagedWebACLV2-?FMS-.*  can match
```
FMManagedWebACLV2-Embargo-US-WEST-2-1656966510987
FMManagedWebACLV2Embargo-US-WEST-2-1656876510999
```

AWS Firewall Manager team changed their WebACL naming convention (added '-' dash) on May 2021.
```
From: FMManagedWebACLV2{WAF_Policy_Name}-{UTC in milliseconds}
To: FMManagedWebACLV2-{WAF_Policy_Name}-{UTC in milliseconds}
```
So we observed any AWS accounts created before May, 2021, we see WebACL name with no dash and accounts that created after WebACL name is with dash.

In this case, we want to cover above two WebACL names using regex "-?".
In glob "?" matching to one character only so it has limitation. I think using regex is more flexible to check WebACL names for various use cases.

Reply:
@cahn1 If we were using glob matching, then in your example, we would probably just do
```
*FMManagedWebACLV2*FMS-.*
```
It'll actually be simpler. However, the downside is that it's not as exact and flexible as regex. I'm leaning toward regex as well in this case. @kapilt

Reply:
@kapilt during the community meeting, you said that we don't lose any precision using fnmatch instead of regex. Is that true? How would you handle the use case @cahn1 has mentioned above? We want to match
`FMManagedWebACLV2{WAF_Policy_Name}-{UTC in milliseconds}`
and
`FMManagedWebACLV2-{WAF_Policy_Name}-{UTC in milliseconds}`

To my best knowledge, fnmatch doesn't support matching zero or one character. We would have to end up using `*`, which means it will match more than what we need. For example, if the matching pattern is `FMManagedWebACLV2*WAF_POLICY_NAME-.*`, we would end up matching things like

`FMManagedWebACLV2---------WAF_POLICY_NAME-1656876510999 `

Reply:
checking out above @kapilt, are we good to move forward with existing implementation on this one?

Reply:
sounds good.